apps/design/backend/src/app.ts

```diff
@@ -330,20 +330,23 @@ function buildApi({ workspace, translator }: AppContext) {
         ballotStrings,
       };
 
-      const extraProps: NhPrecinctSplitOptions = {};
-      if (input.splitId) {
-        const precinct = assertDefined(
-          precincts.find((p) => p.id === input.precinctId)
+      const precinct = find(precincts, (p) => p.id === input.precinctId);
+      let extraProps: NhPrecinctSplitOptions = {};
+      if (hasSplits(precinct)) {
+        const ballotStyle = find(
+          ballotStyles,
+          (bs) => bs.id === input.ballotStyleId
         );
-        // Check for type safety. We expect this precinct to have splits if splitId is provided.
-        if (hasSplits(precinct)) {
-          const split = assertDefined(
-            precinct.splits.find((s) => s.id === input.splitId)
-          );
-          extraProps.electionTitleOverride = split.electionTitleOverride;
-          extraProps.clerkSignatureImage = split.clerkSignatureImage;
-          extraProps.clerkSignatureCaption = split.clerkSignatureCaption;
-        }
+        const { splitId } = find(
+          ballotStyle.precinctsOrSplits,
+          (p) => p.precinctId === input.precinctId
+        );
+        const split = find(precinct.splits, (s) => s.id === splitId);
```

Comment:
Consider factoring out to a helper like `getSplitForBallotStyleAndPrecinct`? I think it'd be helpful to readers to encapsulate the logic.

Reply:
Good idea, doing this in the follow up PR that refactors this code